src/integration/repair/repair_and_replication_test.go

```diff
@@ -0,0 +1,107 @@
+// +build cluster_integration
+//
+// Copyright (c) 2021  Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package repair
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	dbcfg "github.com/m3db/m3/src/cmd/services/m3dbnode/config"
+	dbclient "github.com/m3db/m3/src/dbnode/client"
+	"github.com/m3db/m3/src/dbnode/environment"
+	"github.com/m3db/m3/src/integration/resources"
+	"github.com/m3db/m3/src/integration/resources/inprocess"
+)
+
+func TestRepairAndReplication(t *testing.T) {
+	cluster1, cluster2, closer := testSetup(t)
+	defer closer()
+
+	RunTest(t, cluster1, cluster2)
+}
+
+func testSetup(t *testing.T) (resources.M3Resources, resources.M3Resources, func()) {
+	fullCfgs1 := getClusterFullConfgs(t)
+	fullCfgs2 := getClusterFullConfgs(t)
+
+	setRepairAndReplicationCfg(&fullCfgs1, "cluster-2", &fullCfgs2.EnvConfig)
```

Comment:
Hmm, it's really unfortunate we have to do this preprocessing of the configs because we need to get the envconfig so that the clusters can point at each other. I don't think I did a good job of thinking through this use case beforehand. Maybe we can sync up and some point today and brainstorm ideas around this? What we have here now will work, but is a little complicated to set up. Would be nice if we could have all this driven by the config without doing extra work in the test.